### PR TITLE
chore(UI): downsizing overview

### PR DESCRIPTION
**Pull request by miguel-crespo-fdc**
Removed top level application and Environment Application from the overview.

Ref: SRX-FIC65Q

### CODE REVIEW

File: pkg/db/db.go [hunk 1]
```diff
@@ -1979,7 +1976,10 @@ func processAllLatestDeploymentsForApp(rows *sql.Rows) (map[string]Deployment, e
 			}
 			return nil, fmt.Errorf("Error scanning deployments row from DB. Error: %w\n", err)
 		}
-
+		err = json.Unmarshal(([]byte)(jsonMetadata), &curr.Metadata)
+		if err != nil {
+			return nil, fmt.Errorf("Error during json unmarshal in deployments. Error: %w. Data: %s\n", err, jsonMetadata)
+		}
```
Comment (miguel-crespo-fdc) [inline pkg/db/db.go:1982]: For some reason the metadata wasnt getting unamarshalled

File: services/frontend-service/src/ui/utils/store.tsx [hunk 2]
```diff
@@ -59,12 +58,16 @@ export interface DisplayLock {
 }
 
 export const displayLockUniqueId = (displayLock: DisplayLock): string =>
-    'dl-' + displayLock.lockId + '-' + displayLock.environment + '-' + displayLock.application;
+    'dl-' +
+    displayLock.lockId +
+    '-' +
+    displayLock.environment +
+    '-' +
+    (displayLock.application ? displayLock.application : displayLock.team);
```
Comment (miguel-crespo-fdc) [inline services/frontend-service/src/ui/utils/store.tsx:66]: Was broken for team locks

File: services/rollout-service/pkg/versions/versions.go [hunk 3]
```diff
@@ -98,27 +98,21 @@ func (v *versionClient) GetVersion(ctx context.Context, revision, environment, a
 }
 
 // Tries getting the version from cache
-func (v *versionClient) tryGetVersion(ctx context.Context, revision, environment, application string) (*VersionInfo, error) {
-	var overview *api.GetOverviewResponse
-	entry, ok := v.cache.Get(revision)
+func (v *versionClient) tryGetVersion(environment, application string) (*VersionInfo, error) {
```
Comment (miguel-crespo-fdc) [inline services/rollout-service/pkg/versions/versions.go:101]: Previously, we hand't deleted the Environment Application information from the overview, so this function still worked. As we deleted it, now we were forced to rewrite this function

File: services/frontend-service/pkg/handler/commit_deployments_test.go [hunk 4]
```diff
@@ -83,7 +83,7 @@ func TestHandleCommitDeployments(t *testing.T) {
 			inputTail:          "123456/",
 			failGrpcCall:       false,
 			expectedStatusCode: http.StatusOK,
-			expectedResponse:   "{\"deploymentStatus\":{\"app1\":{\"deploymentStatus\":{\"dev\":\"DEPLOYED\", \"prod\":\"UNKNOWN\", \"stage\":\"PENDING\"}}}}\n",
+			expectedResponse:   "{\"deploymentStatus\":{\"app1\":{\"deploymentStatus\":{\"dev\":\"DEPLOYED\",\"prod\":\"UNKNOWN\",\"stage\":\"PENDING\"}}}}\n",
```
Comment (miguel-crespo-fdc) [inline services/frontend-service/pkg/handler/commit_deployments_test.go:86]: I know this is here again.... But I can't get the pipeline to pass without changing this test.

File: pkg/db/overview_test.go [hunk 5]
```diff
@@ -67,48 +68,54 @@ func makeTestStartingOverview() *api.GetOverviewResponse {
...
-								Name:    "test",
-								Version: 1,
-								DeploymentMetaData: &api.Environment_Application_DeploymentMetaData{
-									DeployAuthor: "testmail@example.com",
-									DeployTime:   "1",
-								},
-								Team: "team-123",
-							},
-						},
+						//Applications: map[string]*api.Environment_Application{
```
Comment (sven-urbanski-freiheit-com) [inline pkg/db/overview_test.go:71]: reference to the story is missing here

File: services/cd-service/pkg/service/overview_test.go [hunk 6]
```diff
@@ -169,17 +333,27 @@ func TestOverviewService(t *testing.T) {
 				if len(resp.EnvironmentGroups) != expectedEnvs {
 					t.Errorf("expected %d environmentGroups, got %q", expectedEnvs, resp.EnvironmentGroups)
 				}
-				testApp := resp.Applications["test"]
-				releases := testApp.Releases
+				app, err := svc.GetAppDetails(ctx, &api.GetAppDetailsRequest{AppName: "test"})
+				if err != nil {
+					t.Errorf("Error fetching information for app test")
```
Comment (sven-urbanski-freiheit-com) [inline services/cd-service/pkg/service/overview_test.go:338]: (nitpick)
```suggestion
					t.Errorf("Error fetching information for app test: %v", err)
```

File: services/cd-service/pkg/service/overview_test.go [hunk 7]
```diff
@@ -188,24 +362,34 @@ func TestOverviewService(t *testing.T) {
 				if releases[0].PrNumber != "" {
 					t.Errorf("Release should not have PR number")
 				}
-				testApp = resp.Applications["test-with-incorrect-pr-number"]
+
+				app, err = svc.GetAppDetails(ctx, &api.GetAppDetailsRequest{AppName: "test-with-incorrect-pr-number"})
+				if err != nil {
+					t.Errorf("Error fetching information for app test")
```
Comment (sven-urbanski-freiheit-com) [inline services/cd-service/pkg/service/overview_test.go:368]: same here

File: services/frontend-service/pkg/handler/commit_deployments_test.go (hunk 4 above)
Comment (sven-urbanski-freiheit-com) [reply services/frontend-service/pkg/handler/commit_deployments_test.go:86]: ok, let's ignore it for this PR

File: services/frontend-service/src/ui/utils/store.tsx [hunk 8]
```diff
@@ -1014,16 +939,16 @@ export const useReleaseOptional = (application: string, env: Environment): Relea
...
-    [
-        ...new Set(
-            Object.values(useEnvironments())
-                .filter((env) => env.applications[application])
-                .map((env) => env.applications[application].version)
-        ),
-    ]
-        .sort((a, b) => b - a)
-        .filter((version) => version !== 0); // 0 means "not deployed", so we filter those out
+// export const useDeployedReleases = (application: string): number[] =>
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/src/ui/utils/store.tsx:942]: remove dead code

**Review by sven-urbanski-freiheit-com — APPROVED** [review]

**Pull request merged**